File: TypeScript/643. Maximum Average Subarray I.ts
// noinspection JSUnusedGlobalSymbols
const findMaxAverage = (nums: number[], k: number): number => {
  let left = 0
  let right = 0
  let average = -Infinity
  let sum = 0

  while (right < nums.length) {
    const window = right - left + 1

    sum += nums[right]

    if (window == k) {
      average = Math.max(sum / k, average)

      sum -= nums[left]
      left++
    }

    right++
  }

  return average
}

File: Go/643. Maximum Average Subarray I.go
package main

import "math"

//goland:noinspection GoUnusedFunction
func findMaxAverage(nums []int, k int) float64 {
	left, right := 0, 0
	average := math.Inf(-1)
	var sum float64

	for right < len(nums) {
		window := right - left + 1

		sum += float64(nums[right])

		if window == k {
			average = max(sum/float64(k), average)

			sum -= float64(nums[left])
			left++
		}

		right++
	}

	return average
}
